### PR TITLE
Ore Stacking

### PR DESCRIPTION
**Pull request by silicons**
Ores can now be stacked to 50. At that point they'll be weight class huge (5). Ore satchels now can hold weight class 5 to compensate, as well as having a 16 combined weight class with 8 slots to compensate.

### CODE REVIEW

**Comment by ShizCalev** [comment]
conflict.

**Comment by Xhuis** [comment]
Expecting at least 150 hours of testing on this.

**Comment by Bawhoppen** [comment]
How does this work with mined ore? Will it look and feel more boring by just having stacks drop now instead?

**Comment by AnturK** [comment]
Please add stack amount sprite variations if you're doing this.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
What antur said

**Comment by KorPhaeron** [comment]
@xhuis please keep your grudge against kevinz off the repo and stop leaving garbage comments with no content on his PRs

**Comment by Xhuis** [comment]
I think you're taking this out of proportion.

**Comment by Bawhoppen** [comment]
Kor IS the proportion.

**Comment by Bawhoppen** [comment]
Sorry. I know that doesn't make sense, but it was really funny in my head for that exact reason

File: code/modules/mining/satchel_ore_boxdm.dm [hunk 1]
```diff
@@ -41,18 +41,21 @@
...
-		oretypes |= O.type
-	for(var/i in oretypes)
-		var/obj/item/ore/T = locate(i) in contents
-		dat += "[capitalize(T.name)]: [count_by_type(contents, T.type)]<br>"
+	var/list/assembled = list()
+	for(var/obj/item/stack/ore/O in src)
+		if(!assembled[O.type])
+			assembled[O.type] = O.amount
+		else
+			assembled[O.type] += O.amount
```
Comment (ninjanomnom) [inline code/modules/mining/satchel_ore_boxdm.dm]: skip the if else just do `assembled[O.type] += O.amount`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/cargo/exports/materials.dm [hunk 2]
```diff
@@ -22,8 +22,8 @@
 	if(istype(I, /obj/item/stack))
 		var/obj/item/stack/S = I
 		amount *= S.amount
-	else if(istype(I, /obj/item/ore))
-		amount *= 0.8 // Station's ore redemption equipment is really goddamn good.
+		if(istype(I, /obj/item/stack/ore))
+			amount *= 0.8 // Station's ore redemption equipment is really goddamn good.
```
Comment (ninjanomnom) [inline code/modules/cargo/exports/materials.dm]: for a total of `amount * S.amount * 0.8` ?

Comment (kevinz000) [reply code/modules/cargo/exports/materials.dm]: yes, per piece * pieces in stack * ore multiplier
unless it doesn't work this way, kek.

**Review by ninjanomnom — APPROVED** [review]

**Comment by Bawhoppen** [comment]
HEY, @kevinz000, You never addressed my issue! This isn't just me who cares about it either! Go ask MSO!

**Comment by Bawhoppen** [comment]
To clarify for anyone who is confused, the issue is that this makes everything look much worse and less flavorful in after mining ore lavaland, which is the basic reason we hadn't done this sooner.

**Comment by vuonojenmustaturska** [comment]
@SgtxRootbeer bad news for you, I'm pretty sure the bomb cap on lavaland is absolutely never getting removed. 

If you want to do it, you can remove the bombcap on your locally hosted server, that way you're not crashing a server with 80 other people on it.

**Pull request merged**